src/instructlab/config/init.py

```diff
@@ -209,6 +145,118 @@ def init(
+    arch_family_processors: dict[str, list[list[str]]] = {}
+    for dirpath, _dirnames, filenames in os.walk(DEFAULTS.SYSTEM_PROFILE_DIR):
+        for filename in filenames:
+            system_profile_files.append(os.path.join(dirpath, filename))
+            arch_family_processor = os.path.relpath(
+                os.path.join(dirpath, filename), DEFAULTS.SYSTEM_PROFILE_DIR
+            ).split("/", 3)
+            arch_family_processors.setdefault(arch_family_processor[0], []).append(
+                arch_family_processor
+            )
+
+    click.echo(
+        click.style(
+            "Please choose a system profile to use.\nSystem profiles apply to all parts of the config file and set hardware specific defaults for each command.",
+            fg="green",
+        )
+    )
+    # print info like APPLE, AMD, INTEL and have them select
+    click.echo(
+        click.style(
+            "First, please select the hardware vendor your system falls into",
+            bg="blue",
+            fg="white",
+        )
+    )
+    keys = list(arch_family_processors.keys())
+    for idx, key in enumerate(keys, 1):
+        print(f"[{idx}] {key.upper()}")
+
+    system_profile_selection = click.prompt(
+        "Enter the number of your choice",
+        type=int,
+        default=0,
+    )
+
+    # now print all choices in the selected hw vendor and have user choose
+    if 1 <= system_profile_selection <= len(keys):
+        key = keys[system_profile_selection - 1]
+        click.echo(f"You selected: {key.upper()}")
+        click.echo(
+            click.style(
+                "Next, please select the specific hardware configuration that most closely matches your system.",
+                bg="blue",
+                fg="white",
+            )
+        )
+        click.echo("[0] No system profile")
+        i = 1
+        for arch_family_processor in arch_family_processors[key]:
+            # the following logic is specifically for printing the name
+            # we still want to preserve the arch_family_processor for when we open the file
+            # if the last entry has an _, split that out. This follows the format like m2_max.yaml, we just want max.
+            printed_arch_family_processor = copy(arch_family_processor)
+            if (
+                "_"
+                in printed_arch_family_processor[len(printed_arch_family_processor) - 1]
+            ):
+                printed_arch_family_processor[
+                    len(printed_arch_family_processor) - 1
+                ] = printed_arch_family_processor[
+                    len(printed_arch_family_processor) - 1
+                ].split("_")[1]
+            # remove the .yaml so now we have something like "apple m2 pro"
+            printed_arch_family_processor[len(printed_arch_family_processor) - 1] = (
+                printed_arch_family_processor[
+                    len(printed_arch_family_processor) - 1
+                ].split(".")[0]
+            )
```

Comment:
Can we just use regexp with a detailed comment to explain what is going on here
this seems unnecessarily verbose

```suggestion
            # the following logic is specifically for printing the name
            # we still want to preserve the arch_family_processor for when we open the file
            # if the last entry has an _, split that out. This follows the format like m2_max.yaml, we just want max.
            # Copy for preservation when working with printed names
            printed_arch_family_processor = copy(arch_family_processor)

            # Process the last element, extracting text after "_" if present, removing ".yaml"
            printed_arch_family_processor[-1] = re.sub(r'_(\w+)\.yaml$|\.yaml$', r' \1', printed_arch_family_processor[-1])
```

Reply:
thanks for this, don't know why I didn't think of this :sweat_smile: